### PR TITLE
doctor: Check for branch of core tap (take 2)

### PR DESCRIPTION
**Pull request by BenMusch**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Fixes the issues with https://github.com/Homebrew/brew/pull/3089 -- the `git_head` returns a commit sha.

Pulls the changes from https://github.com/Homebrew/brew/pull/3094 without the refactor to GitClient, to get the easy win merged first

### CODE REVIEW

File: Library/Homebrew/diagnostic.rb [hunk 1]
```diff
@@ -785,6 +785,19 @@ def check_coretap_git_origin
               git -C "#{coretap_path}" remote set-url origin #{Formatter.url("https://github.com/Homebrew/homebrew-core.git")}
           EOS
         end
+
+        return if ENV["CI"] || ENV["JENKINS_HOME"]
+
+        branch = coretap_path.git_branch
+        return if branch.nil? || branch =~ /master/
```
Comment (MikeMcQuaid) [inline Library/Homebrew/diagnostic.rb]: Should probably be `return if branch != "master"`  as `foo-master-bar` is an invalid branch name.

File: Library/Homebrew/extend/git_repository.rb [hunk 2]
```diff
@@ -36,6 +36,15 @@ def git_last_commit
     end
   end
 
+  def git_branch
+    return unless git? && Utils.git_available?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/git_repository.rb]: Make this an `if` or two statements as `unless &&` is hard to follow.

**Comment by MikeMcQuaid** [comment]
I fixed a `brew style` CI failure for you.

**Comment by MikeMcQuaid** [comment]
Thanks @BenMusch!

**Comment by BenMusch** [comment]
No problem! Sorry for not being super responsive, but I'm still happy to fix any bugs about this if they pop up
